aea/registries/base.py

```diff
@@ -243,8 +243,18 @@ def _add_contract(self, contract_directory: Path):
             open(contract_directory / DEFAULT_CONTRACT_CONFIG_FILE)
         )
 
-        # instantiate the protocol manager.
-        contract = Contract(contract_config.public_id, contract_config)
+        contract_spec = importlib.util.spec_from_file_location(
+            "contracts", contract_directory / "contract.py"
+        )
+
+        contract_module = importlib.util.module_from_spec(contract_spec)
```

Comment:
this should be in a try: except: block as the `SomeContract` class might not be available.

Reply:
Should we consider exiting in case we don't find the file ? 

>         try:
            contract_spec = importlib.util.spec_from_file_location(
                "contracts", contract_directory / "contract.py"
            )
            contract_module = importlib.util.module_from_spec(contract_spec)
        except FileNotFoundError:
            logger.warning("File not found.")
            sys.exit(1)

Reply:
Yes you can do that. In this case also add the try and similar behaviour for the other registries.